### ISSUE TITLE
Feature: Keep the scroll position of the preview pane in sync with what's being edited

### ISSUE THREAD

**Issue by papandreou** [netlify/netlify-cms#1466]
**- Do you want to request a *feature* or report a *bug*?**

*feature*

**- What is the current behavior?**

When you're authoring content, the preview pane either has to be scrolled separately or be in the "autoscroll" mode, which currently seems to mean "scroll the preview pane as many percent down as the content pane". When many individually collapsed and expanded editor components are involved, the autoscroll strategy most often scrolls to a place that's unrelated to what's being edited, so the manual scroll mode is the only usable option.

I'd share screenshots of my current setup, but I'm not at liberty to, unfortunately.

**- What is the expected behavior?**

That the preview pane automatically scrolls into view the preview of the content or component that's actually being edited.

I'm unsure about whether this is already possible without direct core support by using off-the-shelf react components and existing netlify-cms hooks. Pointers would be appreciated :)

**Comment by wraybowling** [comment]
This is something that I've also noticed, know a good bit about, and could possibly help solve. The resolution is going to be to keep track of just about everything related to focus, cursor position, and scrollTop positions, the heights of the two blocks that are related, and the heights of their scrollable viewports. Once you get all of that, it's very easy to write logic about whether the scroll position should move up or down in either direction.

**Comment by erquhart** [comment]
Agreed! Feel free to play around with it, we can answer any questions that might come up here or in [Gitter](https://gitter.im/netlify/netlifycms).

**Comment by zvimoe** [comment]
I'm using  **custom previews,**  and thought of two possible solutions for this issue 

**Option 1**
Adding a fifth prop to a preview component, that will trigger the 'focus' Function on a widget of a specific element  this could enable you to click on the preview element to focus on it's widget and scroll there 

**for example** 

```
h('div', {'onClick':()=>{this.props.focusOnWidget('body')}}, this.props.widgetFor('body'))
```
**Option 2**
Adding a Function to set a custom preview prop in the custom **widget controller** similar to```setActiveStyle()``` and be sent to the custom preview to be used on the widget itself,
then anyone could create there own custom functionality 
**for example**
```
const focusOnWidget = (whatever)=>{
      this.element.focus()
     /* and use whatever for whatever */
}

setCustomPreviewProp('focusOnWidget',this.focusOnWidget })
```
* **Option 2** would give the ability to mix the inputs in with the preview like in wordPress etc. I mean if you could pass a function to the preview then you could pass the widget to the preview as well.
It could be amazing of someone could add add it
